### PR TITLE
{Packaging} Install pip in Homebrew virtual env

### PR DESCRIPTION
**Pull request by bebound**
**Description**

Homebrew does not install pip when create venv in https://github.com/Homebrew/brew/pull/15792, adding `without_pip: false` to install pip.

`pip` also installs `setuptools`, which is also required in namespace package:
```
linuxbrew@794b47da48b2:/down$ az -v
Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/opt/python@3.10/lib/python3.10/runpy.py", line 187, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/home/linuxbrew/.linuxbrew/opt/python@3.10/lib/python3.10/runpy.py", line 110, in _get_module_details
    __import__(pkg_name)
  File "/home/linuxbrew/.linuxbrew/Cellar/azure-cli/2.51.0/libexec/lib/python3.10/site-packages/azure/__init__.py", line 1, in <module>
    ﻿__import__('pkg_resources').declare_namespace(__name__)
ModuleNotFoundError: No module named 'pkg_resources'
```

Related issue:
- https://github.com/Azure/azure-cli/issues/27047
- https://github.com/Azure/azure-cli/pull/27196

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Packaging

File: scripts/release/homebrew/docker/formula_generate.py [hunk 1]
```diff
@@ -126,6 +126,11 @@ def update_formula() -> str:
     upstream_sha = compute_sha256(HOMEBREW_UPSTREAM_URL)
     text = re.sub('sha256 ".*"', 'sha256 "{}"'.format(upstream_sha), text, 1)
     text = re.sub('.*revision.*\n', '', text, 1)  # remove revision for previous version if exists
+
+    # include pip when creating venv, see https://github.com/Azure/azure-cli/issues/27047
```
Comment (jiasli) [inline scripts/release/homebrew/docker/formula_generate.py:130]: It the link incorrect? https://github.com/Azure/azure-cli/issues/27047 doesn't seem to be related to `pip` in a venv.

Comment (bebound) [reply scripts/release/homebrew/docker/formula_generate.py]: Update to homebrew pr link: https://github.com/Homebrew/brew/pull/15792

File: scripts/release/homebrew/docker/formula_generate.py [hunk 2]
```diff
@@ -126,6 +126,11 @@ def update_formula() -> str:
     upstream_sha = compute_sha256(HOMEBREW_UPSTREAM_URL)
     text = re.sub('sha256 ".*"', 'sha256 "{}"'.format(upstream_sha), text, 1)
     text = re.sub('.*revision.*\n', '', text, 1)  # remove revision for previous version if exists
+
+    # include pip when creating venv, see https://github.com/Homebrew/brew/pull/15792
+    # this can be removed after Homebrew merges our PR
+    text = re.sub('system_site_packages: false', 'system_site_packages: false, without_pip: false', text, 1)
```
Comment (jiasli) [inline scripts/release/homebrew/docker/formula_generate.py:132]: As `formula_generate.py` always updates the formula on Homebrew side, it requires a weird process that we always need 2 separate PRs:

1. one to make the change,
2. once the change is merged to Homebrew side's formula, one without the change

An alternative I can think of is to directly submit a PR to https://github.com/Homebrew/homebrew-core/blob/master/Formula/a/azure-cli.rb

**Comment by jiasli** [comment]
https://github.com/Homebrew/brew/pull/15792#issuecomment-1675276254 made a very good point.

- Without `without_pip: false`, our formula breaks on new brew because `pip` is missing
- With `without_pip: false`, our formula breaks on old brew because it doesn't recognize `without_pip: false`

File: scripts/release/homebrew/docker/formula_generate.py (hunk 2 above)
Comment (bebound) [reply scripts/release/homebrew/docker/formula_generate.py:132]: We have to manually fix here as Homebrew pipeline is broken now.
Creating PR to homebrew is more convenient.

**Review by jiasli — APPROVED** [review]
